src/libs/ajax/ajax-common.ts

```diff
@@ -70,20 +70,24 @@ const isUnauthorizedResponse = (error: unknown): boolean => error instanceof Res
 export const withRetryAfterReloadingExpiredAuthToken =
   (wrappedFetch: FetchFn): FetchFn =>
   async (resource: RequestInfo | URL, options?: RequestInit): Promise<Response> => {
+    const preRequestAuthToken = getAuthToken();
     const requestHasAuthHeader = _.isMatch(authOpts(), options as object);
     try {
       return await wrappedFetch(resource, options);
     } catch (error) {
       if (isUnauthorizedResponse(error) && requestHasAuthHeader) {
         const reloadedAuthTokenState: AuthTokenState = await loadAuthToken();
+        const postRequestAuthToken = getAuthToken();
         if (reloadedAuthTokenState.status === 'success') {
           sendRetryMetric();
           const optionsWithNewAuthToken = _.merge(options, authOpts());
           return await wrappedFetch(resource, optionsWithNewAuthToken);
         }
-        const signOutCause: SignOutCause =
-          reloadedAuthTokenState.status === 'expired' ? 'expiredRefreshToken' : 'errorRefreshingAuthToken';
-        signOut(signOutCause);
+        if (preRequestAuthToken === postRequestAuthToken) {
```

Comment:
This seems surprising. We're checking if we actually loaded a new auth token after `loadAuthToken` failed? I'm wondering if it would make sense to coordinate calls to `loadAuthToken`? And also maybe move the `signOut` call into `loadAuthToken`?

Reply:
So, one of the things I have observed in mixpanel has been that several request go out. Then one of those requests comes back with a 401 and then fails to refresh for whatever reason. 

Then the user is signed out. (this will revoke their access and refresh token as part of that flow)
Then the user signs back in.
Then one of the other requests also comes back with a 401 and since the refresh token is already revoked, the refresh will fail.
Then the user is signed out again.

This PR is aiming to remove sign outs from requests which happened before the user was initially signed out.

In the case where the token fails to refresh, the token will not have changed as a result of a single call to this function. So what this aims to do is make sure that if a sign out request is going to be triggered that a new token as not already been issued.

Reply:
Since this code is unintuitive, a comment explaining why this is done would be nice.